pex/pip/runtime_patches.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# Various runtime patches applied to Pip to work around issues or else bend Pip to Pex's needs.
+
+import os
+import runpy
+
+# N.B.: The following environment variables are used by the Pex runtime to control Pip and must be
+# kept in-sync with `tool.py`.
+skip_markers = os.environ.pop("_PEX_SKIP_MARKERS", None)
```

Comment:
I extracted these 4 runtime control env var accesses into this block and added some type ignores for the pip imports inline below; otherwise, this file was a straight move from the source code string in Pip with `tox -eformat-run` applied.